lib/bolt/pal/yaml_plan/parameter.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Bolt
+  class PAL
+    class YamlPlan
+      class Parameter
+        attr_reader :name, :value, :type_expr
+
+        PARAMETER_KEYS = Set['type', 'default', 'description']
+
+        def initialize(param, definition)
+          definition ||= {}
+          validate_param(param, definition)
+
+          @name = param
+          @value = definition['default']
+          @type_expr = Puppet::Pops::Types::TypeParser.singleton.parse(definition['type']) if definition['type']
+        end
+
+        def validate_param(param, definition)
+          unless param.is_a?(String) && param.match?(Bolt::PAL::YamlPlan::VAR_NAME_PATTERN)
+            raise Bolt::Error.new("Invalid parameter name #{param.inspect}", "bolt/invalid-plan")
+          end
+
+          definition_keys = definition.keys.to_set
+          unless PARAMETER_KEYS.superset?(definition_keys)
+            invalid_keys = definition_keys - PARAMETER_KEYS
+            raise Bolt::Error.new("Plan parameter #{param.inspect} contains illegal key(s)" \
+                                  " #{invalid_keys.to_a.inspect}",
+                                  "bolt/invalid-plan")
+          end
+        end
+
+        def captures_rest
```

Comment:
What does this do?

Reply:
I'm not sure - it was already part of the Parameter struct. @nicklewis might know

Reply:
This is necessary to let our fake `Parameter` class masquerade as a real Puppet parameter. I think `captures_rest` is basically equivalent to a varargs type parameter in a Puppet function. Plans don't actually use positional arguments, so it's not a meaningful concept for them.